tests/sessions_tests/tests.py

```diff
@@ -346,6 +327,11 @@ def test_decode_failure_logged_to_security(self):
                 # The failed decode is logged.
                 self.assertIn('Session data corrupted', cm.output[0])
 
+    def test_decode_serializer_exception(self):
+        signer = TimestampSigner(salt=self.session.key_salt)
+        encoded = signer.sign(b'invalid data')
+        self.assertEqual(self.session.decode(encoded), {})
```

Comment:
I added this test to cover catching exception from serializers in [decode()](https://github.com/django/django/commit/8d25fa4d80749ee74097108400bf63044a2a78d0#diff-3a6c11bbe36a0e6927f71ad8d669f0021897ba73768ee41073a318a12e11c3d1R103-R105).